### PR TITLE
Rewrite logic for when to show moderation buttons next to messages

### PR DESCRIPTION
**Pull request by fourtf**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

removes some weird 20 lines of code that are actually two lines of normal code

### CODE REVIEW

**Comment by pajlada** [comment]
I'm still having hard times wrapping my head around the if. I'm not sure if it's double negation + && or if it's the nested conditional - my counter-proposal would be to wrap the logic in a private function, something akin to this:
```cpp
bool TwitchMessageBuilder::shouldAddModerationElements() const {
    if (this->senderIsBroadcaster) {
        return false;
    }

    if (this->tags.value("user-type").toString() == "mod" && this->args.isStaffOrBroadcaster) {
        return false;
    }

    return true;
}
```

We wouldn't shave off any lines, but the logic for moderation buttons is clear, concise enough (7 lines of code vs the previous 20-line mess)

This is not a hill I'm willing to die on, so if that's unacceptable I'd be fine with a comment above the code instead

File: src/providers/twitch/TwitchMessageBuilder.cpp [hunk 1]
```diff
@@ -231,29 +231,8 @@ MessagePtr TwitchMessageBuilder::build()
...
-                {
-                    addModerationElement = false;
-                }
-            }
-        }
-    }
-
-    if (addModerationElement)
+    if (!this->senderIsBroadcaster &&
+        !(this->tags.value("user-type") == "mod" && !args.isStaffOrBroadcaster))
```
Comment (apa420) [inline src/providers/twitch/TwitchMessageBuilder.cpp:235]:
```suggestion
    if (!(this->senderIsBroadcaster ||
        (this->tags.value("user-type") == "mod" && !args.isStaffOrBroadcaster)))
```

Would this fix the confusion @pajlada ?

Comment (pajlada) [reply src/providers/twitch/TwitchMessageBuilder.cpp:235]: No

**Review by pajlada — CHANGES_REQUESTED** [review]
I'm still having hard times wrapping my head around the if. I'm not sure if it's double negation + && or if it's the nested conditional - my counter-proposal would be to wrap the logic in a private function, something akin to this:
```cpp
bool TwitchMessageBuilder::shouldAddModerationElements() const {
    if (this->senderIsBroadcaster) {
        return false;
    }

    if (this->tags.value("user-type").toString() == "mod" && this->args.isStaffOrBroadcaster) {
        return false;
    }

    return true;
}
```

We wouldn't shave off any lines, but the logic for moderation buttons is clear, concise enough (7 lines of code vs the previous 20-line mess)

This is not a hill I'm willing to die on, so if that's unacceptable I'd be fine with a comment above the code instead

**Review by pajlada — APPROVED** [review]

**Pull request merged**